x-pack/plugins/maps/public/connected_components/mb_map/tooltip_control/tooltip_control.js

```diff
@@ -129,25 +242,36 @@ export class TooltipControl extends React.Component {
     }
 
     const targetMbFeature = mbFeatures[0];
-    if (this.props.openTooltips[0]) {
+    if (this.props.openTooltips[0] && this.props.openTooltips[0].features.length) {
       const firstFeature = this.props.openTooltips[0].features[0];
       if (targetMbFeature.properties[FEATURE_ID_PROPERTY_NAME] === firstFeature.id) {
         // ignore hover events when hover tooltip is all ready opened for feature
         return;
       }
     }
-
     const popupAnchorLocation = justifyAnchorLocation(e.lngLat, targetMbFeature);
-    const features = this._getTooltipFeatures(mbFeatures);
+
+    const isLocked = false;
+    const tooltipId = uuid();
+    const features = this._getTooltipFeatures(mbFeatures, isLocked, tooltipId);
+    if (features.length === 0) {
+      return;
+    }
     this.props.openOnHoverTooltip({
       features: features,
       location: popupAnchorLocation,
+      isLocked,
+      id: tooltipId,
     });
   }, 100);
 
   _getMbLayerIdsForTooltips() {
     const mbLayerIds = this.props.layerList.reduce((mbLayerIds, layer) => {
-      return layer.canShowTooltip() ? mbLayerIds.concat(layer.getMbLayerIds()) : mbLayerIds;
+      // tooltips are only supported for vector layers, filter out all other layer types
+      const isVectorLayer = layer.canShowTooltip !== undefined;
```

Comment:
`layer.getType` is not the best type check because its fragile. For example, if new IVectorLayer implementations where added in maps or by other plugins, then the check would fail for those layers. Currently BlendedVectorLayer and VectorLayer are the only implementations but new ones could be added.